pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -352,3 +378,62 @@ func isMaster(machine mapiv1.Machine, client client.Client) bool {
 	}
 	return false
 }
+
+// checkAndDecrement checks if the provided MachineDisruptionBudget allows any disruption
```

Comment:
the func is called `checkAndDecrement` but the comment mentions nothing about decrement. Also what is it checking..? Probably should split in two functions `isMachineDisruptionAllowed` or so which does just that.

Reply:
will update the comment

Reply:
we should split and rename

Reply:
done